modules/openfoam/src/OpenFOAM.f90

```diff
@@ -1004,42 +1005,137 @@ END SUBROUTINE CalcForceActuatorPositionsTower
-   p_OpFM%forceBldRnodes(p_OpFM%nNodesForceBlade) = p_OpFM%BladeLength
-
 
+   ! Tower
    if (p_OpFM%NMappings .gt. p_OpFM%NumBl) then
-      ! tower
       allocate(p_OpFM%forceTwrHnodes(p_OpFM%nNodesForceTower), stat=errStat2);   if (Failed2()) return;
+      ! Compute uniform spacing.
       dRforceNodes = p_OpFM%TowerHeight/(p_OpFM%nNodesForceTower-1)
       do i=1,p_OpFM%nNodesForceTower-1
          p_OpFM%forceTwrHnodes(i) = (i-1)*dRforceNodes
       end do
       p_OpFM%forceTwrHnodes(p_OpFM%nNodesForceTower) = p_OpFM%TowerHeight
    end if
 
+
+
+   ! Blades
+   allocate(cNonUniform(p_OpFM%nNodesForceBlade),stat=errStat2)
+   allocate(sNonUniform(p_OpFM%nNodesForceBlade),stat=errStat2)
+   allocate(pNonUniform(p_OpFM%nNodesForceBlade),stat=errStat2)
+   allocate(pUniform(p_OpFM%nNodesForceBlade),stat=errStat2)
+   allocate(cByS(p_OpFM%nNodesForceBlade),stat=errStat2)
+   allocate(e(p_OpFM%nNodesForceBlade-1),stat=errStat2)
+   allocate(p_OpFM%forceBldRnodes(p_OpFM%nNodesForceBlade), stat=errStat2);   if (Failed2()) return;
+
+   ! Compute uniform spacing.
+   dRforceNodes = p_OpFM%BladeLength/(p_OpFM%nNodesForceBlade-1)
+   do i=1,p_OpFM%nNodesForceBlade-1
+       pUniform(i) = (i-1)*dRforceNodes
+   end do
+   pUniform(p_OpFM%nNodesForceBlade) = p_OpFM%BladeLength
+   p_OpFM%forceBldRnodes(:) = pUniform(:)
+
+   if (p_OpFM%NodeClusterType .eq. 0) then
+       print*, "Using uniform blade force node clustering."
+      !do i = 1, p_OpFM%nNodesForceBlade
+      !    print*, "r(",i,") = ", pUniform(i)
+      !end do
+   end if
+
+
+   ! If non-uniform spacing is called for, compute the spacing.  
+   ! We know that if the spacing is proportional to chord, then at each
+   ! element, ds = a*c, where a is some constant factor across the entire
+   ! blade.  Therefore sum(ds) = sum(a*c) = L, so a is L/sum(c).  Because
+   ! we don't know c at every point exactly beforehand, we have to
+   ! iterate using c from our previous best guess.  We know things have
+   ! converged when a = c/ds is constant along the blade, so that is our
+   ! convergence check.  We take the difference between a = c/ds between
+   ! all neighboring points to see how different they are, and take the
+   ! rms of that error as the convergence measure (eSum).
+   if (p_OpFM%NodeClusterType .eq. 1) then
+
+       ! For chord-based clustering (increase resolution in regions of decreased chord), an iterative solution to the grid spacing is used.
+       ! The initial guess to the spacing is uniform spacing, so start with that.
+       pNonUniform(:) = pUniform(:)
+
+       ! Get the chord at the initial force points.
+       call OpFM_InterpolateForceNodesChord(initOut_AD, p_OpFM, u_OpFM, ErrStat2, ErrMsg2)
+       cNonUniform(1:p_OpFM%nNodesForceBlade) = u_OpFM%forceNodesChord(2:p_OpFM%nNodesForceBlade+1)
+
+       ! Iterate on a chord-based non-uniform spacing.
+       counter = 0
+       e(:) = 1.0e+6
+       eTol = 1.0e-6
+       eSum = sqrt(sum(e(:) * e(:)))
```

Comment:
I would remove the `(:)` when intending to use a whole array. See https://stevelionel.com/drfortran/2008/03/31/doctor-it-hurts-when-i-do-this/. That's from 2008 so it may or may not be relevant today, but I'd say this should be profiled or just remove it to be safe.

Reply:
Good call.  I removed the (:) from line 1097 and 1099.  The behavior is identical to including the (:)